hw/hw2-baseball.sql

```diff
@@ -0,0 +1,67 @@
+-- BDA_696 HW2 Script
+-- Each table will open with all players
+-- and then again for just one player (110029)
+-- for verification
+
+USE baseball;
+
+DROP TABLE IF EXISTS career_avg,
+	season_avg,
+	rolling_avg;
+    
+-- Create career average table
+
+CREATE TABLE career_avg
+	SELECT batter,
+		SUM(Hit) / SUM(atBat) AS bat_avg_career
+	FROM batter_counts JOIN game ON batter_counts.game_id = game.game_id WHERE atBat > 0
+	GROUP BY batter;
+
+SELECT *
+	FROM career_avg;
+	
+SELECT *
+	FROM career_avg WHERE batter = 110029;
+
+-- Create season average table
+
+CREATE TABLE season_avg
+	SELECT batter,
+		YEAR(game.local_date) as season,
+		SUM(Hit) / SUM(atBat) AS bat_avg_season
+	FROM batter_counts JOIN game ON batter_counts.game_id = game.game_id WHERE atBat > 0
+	GROUP BY batter, season
+	ORDER BY batter, season;
+
+SELECT *
+	FROM season_avg;
+	
+SELECT *
+	FROM season_avg WHERE batter = 110029;
+	
+-- Create 100 game rolling average
+
+CREATE TABLE rolling_avg
+	SELECT DATE(game.local_date) AS game_date,
+		batter,
+		(SUM(Hit)
+			OVER (PARTITION BY batter
+				ORDER BY game_date ASC ROWS BETWEEN 101 PRECEDING AND 1 PRECEDING)) 
+			/
+			(SUM(atBat)
+				OVER (PARTITION BY batter
+				ORDER BY game_date ASC ROWS BETWEEN 101 PRECEDING AND 1 PRECEDING))
```

Comment:
Finding a syntax error here when running the code in regard to the "asc rows" sections